ros/src/computing/planning/decision/packages/decision_maker/nodes/decision_maker/decision_maker_node_state_drive.cpp

```diff
@@ -0,0 +1,404 @@
+      tryNextState("on_bus_stop");
+      break;
+    default:
+      tryNextState("on_cruise");
+      break;
+  }
+}
+
+void DecisionMakerNode::updateCruiseState(cstring_t& state_name, int status)
+{
+  if (isEventFlagTrue("received_back_state_waypoint"))
+  {
+    tryNextState("on_back");
+    return;
+  }
+
+  if (current_status_.change_flag == enumToInteger<E_ChangeFlags>(E_ChangeFlags::STRAIGHT))
+  {
+    switch (getSteeringStateFromWaypoint())
+    {
+      case autoware_msgs::WaypointState::STR_LEFT:
+        tryNextState("on_left_turn");
+        break;
+      case autoware_msgs::WaypointState::STR_RIGHT:
+        tryNextState("on_right_turn");
+        break;
+      case autoware_msgs::WaypointState::STR_STRAIGHT:
+        tryNextState("on_straight");
+        break;
+      default:
+        break;
+    }
+  }
+  else
+  {
+    if (current_status_.change_flag == enumToInteger<E_ChangeFlags>(E_ChangeFlags::LEFT))
+    {
+      tryNextState("lane_change_left");
+    }
+    else if (current_status_.change_flag == enumToInteger<E_ChangeFlags>(E_ChangeFlags::RIGHT))
+    {
+      tryNextState("lane_change_right");
+    }
+  }
+}
+
+void DecisionMakerNode::updateFreeAreaState(cstring_t& state_name, int status)
+{
+}
+
+void DecisionMakerNode::updateBusStopState(cstring_t& state_name, int status)
+{
+}
+
+void DecisionMakerNode::updatePullOverState(cstring_t& state_name, int status)
+{
+  publishLampCmd(E_Lamp::LAMP_LEFT);
+}
+
+void DecisionMakerNode::updatePullOutState(cstring_t& state_name, int status)
+{
+  publishLampCmd(E_Lamp::LAMP_RIGHT);
+}
+
+void DecisionMakerNode::entryTurnState(cstring_t& state_name, int status)
+{
+  std::pair<uint8_t, int> get_stopsign = getStopSignStateFromWaypoint();
+  if (get_stopsign.first != 0)
+  {
+    current_status_.found_stopsign_idx = get_stopsign.second;
+    tryNextState("found_stopline");
+    return;
+  }
+  tryNextState("clear");
+}
+
+void DecisionMakerNode::updateLeftTurnState(cstring_t& state_name, int status)
+{
+  publishLampCmd(E_Lamp::LAMP_LEFT);
+}
+
+void DecisionMakerNode::updateStraightState(cstring_t& state_name, int status)
+{
+  publishLampCmd(E_Lamp::LAMP_CLEAR);
+}
+
+void DecisionMakerNode::updateRightTurnState(cstring_t& state_name, int status)
+{
+  publishLampCmd(E_Lamp::LAMP_RIGHT);
+}
+
+void DecisionMakerNode::entryLaneChangeState(cstring_t& state_name, int status)
+{
+  tryNextState("check_target_lane");
+}
+void DecisionMakerNode::updateLeftLaneChangeState(cstring_t& state_name, int status)
+{
+  publishLampCmd(E_Lamp::LAMP_LEFT);
+}
+void DecisionMakerNode::updateRightLaneChangeState(cstring_t& state_name, int status)
+{
+  publishLampCmd(E_Lamp::LAMP_RIGHT);
+}
+
+void DecisionMakerNode::updateCheckLeftLaneState(cstring_t& state_name, int status)
+{
+  /* need safety check function */
+  // static bool is_target_lane_safe = false;
+}
+
+void DecisionMakerNode::updateCheckRightLaneState(cstring_t& state_name, int status)
+{
+  /* need safety check function */
```

Comment:
Future tasks should be specified in the function of pull request or managed as isue.

Reply:
I removed these comments and created issue #1726.